sdk/instance/instance.go

```diff
@@ -21,14 +22,20 @@ type Instance struct {
 	container  container.Container
 	service    *servicesdk.Service
 	instanceDB database.InstanceDB
+
+	endpoint   string
+	engineName string
 }
 
 // New creates a new Instance SDK with given options.
-func New(c container.Container, service *servicesdk.Service, instanceDB database.InstanceDB) *Instance {
+func New(c container.Container, service *servicesdk.Service, instanceDB database.InstanceDB, endpoint string) *Instance {
+	engineName, _, _ := xnet.SplitHostPort(endpoint)
```

Comment:
I don't really like the idea to decompose the name based on the endpoint. It would be better to have the name and port and recompose the endpoint with that in the instance/start file